Makefile

```diff
@@ -226,22 +235,11 @@ $(SECURITY_GO_BINS): depend
 .PHONY: go-build
 go-build: $(PILOT_GO_BINS) $(MIXER_GO_BINS) $(SECURITY_GO_BINS)
 
-# The following are convenience aliases for most of the go targets
-# The first block is for aliases that are the same as the actual binary,
-# while the ones that follow need slight adjustments to their names.
-
-IDENTITY_ALIAS_LIST:=istioctl mixc mixs pilot-agent servicegraph sidecar-injector
-.PHONY: $(IDENTITY_ALIAS_LIST)
-$(foreach ITEM,$(IDENTITY_ALIAS_LIST),$(eval $(ITEM): ${ISTIO_BIN}/$(ITEM)))
-
-.PHONY: istio-ca
-istio-ca: ${ISTIO_BIN}/istio_ca
-
-.PHONY: node-agent
-node-agent: ${ISTIO_BIN}/node_agent
-
-.PHONY: pilot
-pilot: ${ISTIO_BIN}/pilot-discovery
+# The following are convenience aliases for most of the Go targets
+MAPPING_MODULES=pilot.pilot-discovery proxy.pilot-agent istioctl.istioctl sidecar_initializer.sidecar-initializer mixs.mixs mixc.mixc servicegraph.servicegraph node-agent.node_agent istio-ca.istio_ca
```

Comment:
While this PR has been pending the sidecar-initializer was renamed to sidecar-injector so that name in this line needs to be updated.  My personal preference is to head in the direction of all aliases matching the name of the binary (e.g., "sidecar-injector.sidecar-injector").  The main holdout was going to be pilot vs pilot-discovery (after someone straightened out the situation on whether istio-ca and node-agent were to use dashes or underscores).

This change also maps a proxy alias to pilot-agent, which I don't quite understand but I also don't know what pilot-agent is so it could just be my ignorance (the Istio binaries are mostly just black boxes to me).  I do know that Istio has a whole repo called proxy and from that we download the proxy executable to ${ISTIO_BIN}/envoy so I'm concerned it might confuse some people to have an alias called "proxy" that maps to a pilot go executable.  I'll defer to more knowledgable Istio devs on the matter so now I'm just raising this point as food for thought.